### PR TITLE
FEATURE: New sitemap implementation including images

### PR DESCRIPTION
**Pull request by Sebobo**
This change uses a new implementation for the sitemap
to increase rendering speed and optionally allow rendering
of images in the sitemap.

Resolves: #41

### CODE REVIEW

File: Resources/Private/Fusion/Prototypes/XmlSitemapUrl.fusion [hunk 1]
```diff
@@ -1,18 +1,18 @@
 prototype(Neos.Seo:XmlSitemap.Url) < prototype(Neos.Fusion:Component) {
     node = null
-    lastModificationDateTime = ${Date.format(this.node.lastModificationDateTime, 'Y-m-d\TH:iP')}
-    changeFrequency = ${q(this.node).property('xmlSitemapChangeFrequency')}
-    priority = ${q(this.node).property('xmlSitemapPriority') || 0.5}
-
-    @if.isNotShortcut = ${!q(this.node).is('[instanceof Neos.Neos:Shortcut]')}
-    @if.indexable = ${!q(this.node).property('metaRobotsNoindex')}
+    lastModificationDateTime = 'now'
```
Comment (daniellienert) [inline Resources/Private/Fusion/Prototypes/XmlSitemapUrl.fusion]:
```suggestion
    lastModificationDateTime = ${Date.now()}
```

**Comment by Sebobo** [comment]
I would also love feedback regarding extensibility.
Maybe some methods need to be split or changed to allow overriding parts for performance reasons in a custom implementation.
And also some fusion parts which should be better extensible.

**Comment by Sebobo** [comment]
Another note:

It makes sense to soonish refactor the dimension menu in the core to allow using it as a helper to get all node variants for rendering any kind of menu or alternative list.
This way the links could already be generated in the XmlSitemapImplementation and rendered more efficiently.

**Comment by nezaniel** [comment]
> Another note:
> 
> It makes sense to soonish refactor the dimension menu in the core to allow using it as a helper to get all node variants for rendering any kind of menu or alternative list.
> This way the links could already be generated in the XmlSitemapImplementation and rendered more efficiently.

well, we could always use NodeInterface::getOtherNodeVariants, couldn't we?

**Comment by Sebobo** [comment]
> > Another note:
> > It makes sense to soonish refactor the dimension menu in the core to allow using it as a helper to get all node variants for rendering any kind of menu or alternative list.
> > This way the links could already be generated in the XmlSitemapImplementation and rendered more efficiently.
> 
> well, we could always use NodeInterface::getOtherNodeVariants, couldn't we?

I see the problem, that we then have variants but we don't know which ones we should render and which ones are accessible. That is already done in the implementation of the dimensions menu.

**Comment by Sebobo** [comment]
Still have a performance issue, I'm investigating.

**Pull request merged**